### PR TITLE
RPRBLND-1887: Error crashing with wave texture.

### PR DESCRIPTION
**Pull request by DagerD**
### PURPOSE
If object has vast number of faces it consumption a lot of memory and blender hangs.

### EFFECT OF CHANGE
Added warning if object has more than 1 000 000 triangles.

### TECHNICAL STEPS
Added check in objects sync.

### CODE REVIEW

File: src/rprblender/export/object.py [hunk 1]
```diff
@@ -38,6 +38,11 @@ def sync(rpr_context, obj: bpy.types.Object, **kwargs):
 
     log("sync", obj, obj.type)
 
+    if obj.type in ('MESH', 'CURVE', 'FONT', 'SURFACE', 'META') \
+            and (len(obj.data.polygons)) > 1000000:
+        log.warn(f'Found object {obj.name_full} with {len(obj.data.polygons)} faces. '
+                 f'Consider simplifying geometry to less than 1 000 000 faces')
```
Comment (bnagirniak) [inline src/rprblender/export/object.py:44]: I think this overall check should be moved to mesh.py. Also set 1000000 as constant

**Review by bnagirniak — CHANGES_REQUESTED** [review]
Commented in code

File: src/rprblender/export/mesh.py [hunk 2]
```diff
@@ -311,6 +314,11 @@ def sync_visibility(rpr_context, obj: bpy.types.Object, rpr_shape: pyrpr.Shape,
 def sync(rpr_context: RPRContext, obj: bpy.types.Object, **kwargs):
     """ Creates pyrpr.Shape from obj.data:bpy.types.Mesh """
 
+    if obj.type in ('MESH', 'CURVE', 'FONT', 'SURFACE', 'META') \
+            and (len(obj.data.polygons)) > FACES_LARGE_AMOUNT:
+        log.warn(f'Found object {obj.name_full} with {len(obj.data.polygons)} faces. '
+                 f'Consider simplifying geometry to less than 1 000 000 faces')
```
Comment (bnagirniak) [inline src/rprblender/export/mesh.py:320]: change 1000000 to {FACES_LARGE_AMOUNT:,}

**Review by bnagirniak — APPROVED** [review]

File: src/rprblender/export/mesh.py [hunk 3]
```diff
@@ -311,6 +314,11 @@ def sync_visibility(rpr_context, obj: bpy.types.Object, rpr_shape: pyrpr.Shape,
 def sync(rpr_context: RPRContext, obj: bpy.types.Object, **kwargs):
     """ Creates pyrpr.Shape from obj.data:bpy.types.Mesh """
 
+    if obj.type in ('MESH', 'CURVE', 'FONT', 'SURFACE', 'META') \
+            and (len(obj.data.polygons)) > FACES_LARGE_AMOUNT:
+        log.warn(f'Found object {obj.name_full} with {len(obj.data.polygons)} faces. '
```
Comment (bnagirniak) [inline src/rprblender/export/mesh.py:319]: Move this warn to MeshData.init...

**Review by bnagirniak — APPROVED** [review]

**Review by VascoPi — APPROVED** [review]
Tested, work well.

**Review by bnagirniak — APPROVED** [review]
Tested - works good

File: src/rprblender/export/mesh.py [hunk 4]
```diff
@@ -29,6 +29,9 @@
 log = logging.Log(tag='export.mesh')
 
 
+TRIANGLES_LARGE_AMOUNT = 1000000
```
Comment (bsavery) [inline src/rprblender/export/mesh.py:32]: Please rename this to 
"EXCESS_TRIANGLES_THRESHOLD"

**Comment by bsavery** [comment]
This is a random thought.  Is the error coming in RPR or in creating the numpy array?  Is it possible you're overflowing number of max int indices in the array?

**Comment by DagerD** [comment]
> This is a random thought. Is the error coming in RPR or in creating the numpy array? Is it possible you're overflowing number of max int indices in the array?

There is no error, only extremely long syncing of objects with big amount of triangles.

**Pull request merged**